src/overlays/actors/ovl_Dm_Ravine/z_dm_ravine.c

```diff
@@ -6,29 +12,68 @@
 
 void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Destroy(Actor* thisx, GlobalContext* globalCtx);
+void DmRavine_Action(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Update(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
-const ActorInit Dm_Ravine_InitVars = {
-    ACTOR_DM_RAVINE,
-    ACTORCAT_ITEMACTION,
-    FLAGS,
-    OBJECT_KEIKOKU_DEMO,
-    sizeof(DmRavine),
-    (ActorFunc)DmRavine_Init,
-    (ActorFunc)DmRavine_Destroy,
-    (ActorFunc)DmRavine_Update,
-    (ActorFunc)DmRavine_Draw
-};
-*/
+const ActorInit Dm_Ravine_InitVars = { ACTOR_DM_RAVINE,
+                                       ACTORCAT_ITEMACTION,
+                                       FLAGS,
+                                       OBJECT_KEIKOKU_DEMO,
+                                       sizeof(DmRavine),
+                                       (ActorFunc)DmRavine_Init,
+                                       (ActorFunc)DmRavine_Destroy,
+                                       (ActorFunc)DmRavine_Update,
+                                       (ActorFunc)DmRavine_Draw };
+
+void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx) {
+    u8 flag;
+
+    flag = gSaveContext.perm.weekEventReg[0];
+    if (((flag & 0x10) | cREG(0)) != 0) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    THIS->loaded = false;
+    globalCtx->roomContext.unk7A[0] = 1;
+    globalCtx->roomContext.unk7A[1] = 0;
+    THIS->state = 0;
+    Actor_SetScale(thisx, 1.0f);
+    THIS->actionFunc = &DmRavine_Action;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Init.asm")
+void DmRavine_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Ravine_0x80A2E7A0/DmRavine_Destroy.asm")
+// This function is unused, but takes two word-sized arguments
+// `thisx` & `globalCtx` are guesses; but it could be something else
+void DmRavine_Action(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
This should simply be `DmRavine_Destroy`, without any comment. The function is not unused, it just does not have any action in this specific actor.

Reply:
The diff is a bit misleading: this function was previously `func_80A2E838`, not `DmRavine_Destroy` (but the contents of the functions happen to be identical)

Reply:
So `DmRavine_Destroy` is still in the actor?

Reply:
It would probably just be something like DmRavine_DoNothing(DmRavine* this, GlobalContext* globalCtx)

Reply:
Yeah

Reply:
I agree with this function being renamed DmRavine_DoNothing